pkg/libvmi/status/status.go

```diff
@@ -107,3 +119,10 @@ func WithVolumeStatus(volumeStatus v1.VolumeStatus) Option {
 		vmiStatus.VolumeStatus = append(vmiStatus.VolumeStatus, volumeStatus)
 	}
 }
+
+// WithMigrationState sets the migration state
+func WithMigrationState(migrationState v1.VirtualMachineInstanceMigrationState) Option {
```

Comment:
The `MigrationState` field is a pointer, why don't you declare `migrationState` (parameter) as a pointer too? It would avoid the additional copy when calling the function

Reply:
It's fine, I don't think we should be overprotective but up to you.